src/test/rpc/Simulate_test.cpp

```diff
@@ -0,0 +1,819 @@
+        {
+            tx_json = result[jss::tx_json];
+        }
+        else
+        {
+            auto unHexed = strUnHex(result[jss::tx_blob].asString());
+            SerialIter sitTrans(makeSlice(*unHexed));
+            tx_json = STObject(std::ref(sitTrans), sfGeneric)
+                          .getJson(JsonOptions::none);
+        }
+        BEAST_EXPECT(tx_json[jss::TransactionType] == tx[jss::TransactionType]);
+        BEAST_EXPECT(tx_json[jss::Account] == tx[jss::Account]);
+        BEAST_EXPECT(
+            tx_json[jss::SigningPubKey] == tx.get(jss::SigningPubKey, ""));
+        BEAST_EXPECT(
+            tx_json[jss::TxnSignature] == tx.get(jss::TxnSignature, ""));
+        BEAST_EXPECT(tx_json[jss::Fee] == tx.get(jss::Fee, expectedFee));
+        BEAST_EXPECT(
+            tx_json[jss::Sequence] == tx.get(jss::Sequence, expectedSequence));
+    }
+
+    void
+    checkBasicReturnValidity(
+        Json::Value& result,
+        Json::Value& tx,
+        int expectedSequence,
+        XRPAmount expectedFee)
+    {
+        return checkBasicReturnValidity(
+            result, tx, expectedSequence, expectedFee.jsonClipped().asString());
+    }
+
+    void
+    testTx(
+        jtx::Env& env,
+        Json::Value& tx,
+        std::function<void(Json::Value, Json::Value&)> validate,
+        bool testSerialized = true)
+    {
+        Json::Value params;
+        params[jss::tx_json] = tx;
+        validate(env.rpc("json", "simulate", to_string(params)), tx);
+        params[jss::binary] = true;
+        validate(env.rpc("json", "simulate", to_string(params)), tx);
+        validate(env.rpc("simulate", to_string(tx)), tx);
+        validate(env.rpc("simulate", to_string(tx), "binary"), tx);
+        if (testSerialized)
+        {
+            // This cannot be tested in the multisign autofill scenario
+            // It is technically not a valid STObject, so the following line
+            // will crash
+            STParsedJSONObject parsed(std::string(jss::tx_json), tx);
+            auto const tx_blob =
+                strHex(parsed.object->getSerializer().peekData());
+            if (BEAST_EXPECT(parsed.object.has_value()))
+            {
+                Json::Value params;
+                params[jss::tx_blob] = tx_blob;
+                validate(env.rpc("json", "simulate", to_string(params)), tx);
+                params[jss::binary] = true;
+                validate(env.rpc("json", "simulate", to_string(params)), tx);
+            }
+            validate(env.rpc("simulate", tx_blob), tx);
+            validate(env.rpc("simulate", tx_blob, "binary"), tx);
+        }
+    }
+
+    void
+    testParamErrors()
+    {
+        testcase("Test parameter errors");
+
+        using namespace jtx;
+        Env env(*this);
+        Account const alice("alice");
+
+        {
+            // No params
+            Json::Value params = Json::objectValue;
+            auto resp = env.rpc("json", "simulate", to_string(params));
```

Comment:
const for reps here and below

Reply:
Fixed in 1c0b3a6f8ca9b038035689b68971ad5e31ba30df